File: Fibonachi a4a2a0a1a3a5/Without array/Ruby/fib.rb
quantity=gets()
quantity=quantity.to_i
previous=0
current=1
sum=0
i=0
i=i.to_i
result_string=""+current.to_s
while (i<quantity)
	sum=current
	current=current+previous
	previous=sum
	i=i+1
	if ((i/2)*2==i)
		result_string=current.to_s+result_string
	else
		result_string=result_string+current.to_s
	end
end
puts(result_string)

File: Fibonachi a4a2a0a1a3a5/Without array/Java/fib.java
import java.util.Scanner;
public class fib{
	public static void main(String[] args) {
		int i, quantity, current, previous, sum;
		Scanner scan = new Scanner(System.in);
		i=0;
		quantity=scan.nextInt();
		current=1;
		previous=0;
		sum=0;
		String result_string = "" + current;
		while (i<quantity){
			sum=current;
			current=current+previous;
			previous=sum;
			i=i+1;
			if((i/2)*2 == i){
				result_string=current+result_string;}
				else{
				result_string=result_string+current;
			}
		}
		System.out.println(result_string);
	}
}